lib/std/zig/Parse.zig

```diff
@@ -1695,6 +1697,23 @@ fn parseExprPrecedence(p: *Parse, min_prec: i32) Error!Node.Index {
         if (info.prec == banned_prec) {
             return p.fail(.chained_comparison_operators);
         }
+        if (info.prec > 0) {
+            if (tok_tag != prev_op and info.prec == prev_prec) blk: {
+                if (prev_op == .plus and tok_tag == .minus) break :blk;
+                if (prev_op == .minus and tok_tag == .plus) break :blk;
+                if (prev_op == .plus_percent and tok_tag == .minus_percent) break :blk;
+                if (prev_op == .minus_percent and tok_tag == .plus_percent) break :blk;
+                if (prev_op == .plus_pipe and tok_tag == .minus_pipe) break :blk;
+                if (prev_op == .minus_pipe and tok_tag == .plus_pipe) break :blk;
+
+                if (prev_op == .asterisk and tok_tag == .slash) break :blk;
+                if (prev_op == .slash and tok_tag == .asterisk) break :blk;
```

Comment:
IMO it'd be better to allow mixing all operators with the same precedence (except maybe `||` and `**` which could also have their precedences raised) but I'm fine with this too.